File: sdks/typescript/examples/hello-world.ts
/**
 * Hello World — minimal Talos module in TypeScript.
 *
 * Compile: talos compile --language typescript --world minimal-node hello-world.ts
 */
import { talosModule } from "@talos/sdk";

export default talosModule({
  world: "minimal-node",
  run: (data) => {
    const name = (data.name as string) ?? data.input?.toString() ?? "World";
    return {
      greeting: `Hello, ${name}!`,
      source: "typescript-sdk",
    };
  },
});

File: sdks/python/examples/hello_world.py
"""Hello World — minimal Talos module in Python.

Compile: talos compile --language python --world minimal-node hello_world.py
"""

from talos_sdk import talos_module


@talos_module(world="minimal-node")
def run(data: dict) -> dict:
    name = data.get("name", data.get("input", {}).get("name", "World"))
    return {
        "greeting": f"Hello, {name}!",
        "source": "python-sdk",
    }
